### PR TITLE
997 git based alpha builds

### PR DESCRIPTION
**Pull request by tst-kfreiheit**
Parallel development of a release branch and a future release are now possible on the earthenterprise repo.

Steps to enable alpha release development:
1. Create a branch for the current release (i.e. release_5.2.4):

```
git checkout master
git checkout -b release_5.2.4

```

2. On the master branch, add an empty commit and tag it using Major.Minor.Release-Patch format (i.e. 5.2.5-0):

```
git checkout master
git commit --allow-empty -m"5.2.5-0 commit for tag"
```

That's it.  The next commit to master will produce the version 5.2.5-0.alpha, and commits on the release_5.2.4 branch will continue on the 5.2.4 version sequence.

The script earthenterprise/earth_enterprise/src/scons/getversion.py generates the version string, and can be run independently:

# long version (5.2.5-0.alpha.timestamp.hash)
earthenterprise/earth_enterprise/src/scons/getversion.py --long

# short version (5.2.5)
earthenterprise/earth_enterprise/src/scons/getversion.py

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -22,3 +22,4 @@ earth_enterprise/tutorial/FusionTutorial/KML
 /*.files
 /*.includes
 
+.vscode/
```
Comment (tst-dsugar) [inline .gitignore]: Shouldn't this be in the top level .gitignore like before??

Comment (tst-kfreiheit) [reply .gitignore]: I made the change so regardless of which directory you load as your root in vscode, the project will ignore its contents.

I must have opened the source at earthenterprise/earth_enterprise, and saw the extra folder in my untracked files.

This change avoids the issue in all scenarios (including root).

File: earth_enterprise/src/scons/khEnvironment.py [hunk 2]
```diff
@@ -236,149 +236,6 @@ def EmitLongVersionStrfunc(target, backupFile, label):
   return 'EmitLongVersion(%s, %s, %s)' % (target, backupFile, label)
```
Comment (tst-kfreiheit) [inline earth_enterprise/src/scons/khEnvironment.py]: Moved all of the version-related code to its own file, and export only the functions used by khEnvironment (GetVersion, GetLongVersion).

File: earth_enterprise/src/scons/getversion.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+        base = _ReadBackupVersionFile(backupFile)
+        ret = '-'.join([base, _GetDateString()])
+
+  # Append the label, if there is one.
+    if len(label):
+        ret = '.'.join([ret, label])
+
+    return ret
+
+
```
Comment (tst-kfreiheit) [inline earth_enterprise/src/scons/getversion.py]: Everything below is 'not public' implementation detail.  Public interface is above.

File: earth_enterprise/src/scons/getversion.py [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+        return _VersionForTaggedHead()
+    else:
+        return _VersionFromTagHistory()
+
+
+def _IsCurrentCommitTagged():
+    """True if the current commit is tagged, otherwise False"""
+    raw = _GetCommitRawDescription()
+    # If this condition hits, then we are currently on a tagged commit.
+    return (len(raw.split("-")) < 4)
```
Comment (tst-jhaight) [inline earth_enterprise/src/scons/getversion.py]: An alternative way of doing this which would allow a little more flexibility in tag naming: `git describe --exact-match HEAD` will return the tag name if the current commit is tagged and will fail otherwise.  There's nothing wrong with what is here and other places also have the tag scheme built in.

Comment (tst-kfreiheit) [reply earth_enterprise/src/scons/getversion.py]: Thanks -- will leave code as-is, but that's a great tip on git describe.
